src/script/interpreter.h

```diff
@@ -98,7 +98,25 @@ enum
 
 bool CheckSignatureEncoding(const std::vector<unsigned char> &vchSig, unsigned int flags, ScriptError* serror);
 
+class CachedHashes
+{
+public:
+    uint256 hashPrevouts,hashSequence,hashOutputs;
+    bool IsEmpty() const
+    {
+        CachedHashes empty;
+        return (*this) == empty;
+    }
+    bool operator==(const CachedHashes& rhs) const
+    {
+        return hashPrevouts == rhs.hashPrevouts &&
+               hashSequence == rhs.hashSequence &&
+               hashOutputs == rhs.hashOutputs;
+    }
+};
+
 uint256 SignatureHash(const CScript &scriptCode, const CTransaction& txTo, unsigned int nIn, int nHashType, const CAmount& amount, int sigversion);
+uint256 SignatureHash(const CScript &scriptCode, const CTransaction& txTo, unsigned int nIn, int nHashType, const CAmount& amount, int sigversion, CachedHashes& cache);
```

Comment:
wait why does it compiles at all ? my implementation is CachedHashes* not CachedHashes& will fix that. oO